### ISSUE TITLE
Support one-sided gradient?

### ISSUE THREAD

**Issue by mattwescott** [google/jax#3324]
```python
def f(x):
    return jax.scipy.special.logsumexp(jax.numpy.array([np.log(x), 0]))
```

```bash
jax.grad(f)(0.0)
> nan
jax.grad(f)(1e-15)
> 1.0
```

**Comment by mattjj** [comment]
Thanks for the question! Can you take a look at [the "enforcing a differentiation convention" section](https://colab.sandbox.google.com/github/google/jax/blob/master/docs/notebooks/Custom_derivative_rules_for_Python_code.ipynb#scrollTo=V9tHAfrSF1N-) of [the custom derivatives tutorial](https://colab.sandbox.google.com/github/google/jax/blob/master/docs/notebooks/Custom_derivative_rules_for_Python_code.ipynb#scrollTo=LqiaKasFjH82), and let me know if that covers what you're thinking about?
